--- a/src/utils.c
+++ b/src/utils.c
@@ -24,6 +24,7 @@
 
 #include <glib.h>
 #include <glib/gi18n.h>
+#include <gio/gio.h>
 
 #include <locale.h>
 #include <langinfo.h>
@@ -34,6 +35,22 @@
 gboolean
 is_locale_12h(void)
 {
+    const gchar *user_value;
+    GSettings* settings;
+    GVariant * user_variant;
+
+    settings = g_settings_new ("org.gnome.desktop.interface");
+    user_variant = g_settings_get_user_value (settings, "clock-format");
+    user_value = g_variant_get_string (user_variant, NULL);
+    if (user_variant != NULL) {
+        if (strstr(user_value, "12h")) {
+            return TRUE;
+        } else {
+            return FALSE;
+        }
+    }
+
+
     int i;
     static const char *formats_24h[] = {"%H", "%R", "%T", "%OH", "%k", NULL};
     const char* t_fmt = nl_langinfo(T_FMT);
